rwatch/ui/animation/animation.c

```diff
@@ -60,9 +63,14 @@ void _animation_update(Animation *anim)
     if (progress > anim->duration) {
         /* Ok, we're done. */
         if (anim->impl.update)
-            anim->impl.update(anim, ANIMATION_NORMALIZED_MAX);
+            //anim->impl.update(anim, ANIMATION_NORMALIZED_MAX);
         if (anim->impl.teardown)
             anim->impl.teardown(anim);
+        
+        // Free the prop anim
+        if (anim->property_anim)
```

Comment:
Hmm, I wonder if it's possible to have `animation` itself not depend on `property_animation`.  I think that, for instance, this `property_animation_destroy` could be part of an `AnimationImplementation`'s `teardown` routine -- and similarly, the `update` routine, the same.  In particular, I'm kind of thinking that an `AnimationImplementation` should contain an `Animation` -- not the other way around?